keps/sig-network/2999-host-network-pod-ips/README.md

```diff
@@ -0,0 +1,603 @@
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Host-network pods inherit their Pod IPs from the node where they are running. Nodes may change their IPs during their lifecycle, and kubelet updates the node.Status accordingly, however, host-network pods IPs are not updated and keep reporting the old IPs until they are restarted/recreated.
+
+## Motivation
+
+Kubelet reporting the updated IP in the node.Status, but host network pods reporting an IP that no longer exist is misleading and create confusion.
+
+### Goals
+
+* Host-network Pod IPs to reflect the actual IP.
+
+### Non-Goals
+
+* Changes on the kubelet logic to update the node.Status addresses.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+* As a cluster admin I want to use DHCP to configure my nodes networking, and I'd like my Pods to reflect the actual IP of the node without having to restart the node.
+
+### Notes/Constraints/Caveats (Optional)
+
+This is a change on the behavior of host-network pods.
+
+### Risks and Mitigations
+
+Applications running in a host-network pod that depend on the Downward API or the Pod.Status.PodIPs for auto-configuration should consider that it can change. However, most probably these applications would not be working correctly because of that.
```

Comment:
Well, there are two cases here:

1. a node IP is removed (which is what you're talking about)
2. the node's "primary node IP(s)" change in some way but the old primary node IP(s) still exist
   - A. the node goes from single-stack to dual-stack and so starts using dual-stack `podIPs` for new host-network pods.
   - B. the node goes from dual-stack to single-stack and so stops using dual-stack `podIPs`, although presumably the secondary-IP-family node IP still _exists_ even though kubelet is ignoring it.
   - C. the node's `.Status.Addresses` gets reordered in a way that causes a different IP to become primary. (eg, kubelet gets restarted with a different `--node-ip` value, or the user modifies the cloud configuration for the node to, eg, swap two virtual NICs)

In all of the case 2 scenarios, the pod's old IP continues to be at least partially valid/usable.